shared/teams/team/settings/retention/dropdown.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import * as React from 'react'
+import type {RouteProps} from '../../../../route-tree/render-route'
+import PopupMenu, {ModalLessPopupMenu, type MenuItem} from '../../../../common-adapters/popup-menu'
+import {connect, isMobile} from '../../../../util/container'
+
+type OwnProps = RouteProps<
+  {
+    items: Array<MenuItem | 'Divider' | null>,
+  },
+  {}
+> & {
+  navigateUp: () => any,
+}
+
+const mapStateToProps = (state, {routeProps}: OwnProps) => ({
+  items: routeProps.get('items'),
+})
+
+const mapDispatchToProps = (dispatch: Dispatch, {navigateUp}: OwnProps) => ({
+  onHidden: () => dispatch(navigateUp()),
+})
+
+export type Props = {
+  items: Array<MenuItem | 'Divider' | null>,
+  onHidden: () => void,
+}
+
+const RetentionDropdown = (props: Props) => {
+  const {items, onHidden} = props
+  return isMobile ? (
```

Comment:
@akalin this is what i'm talking about re: the dialogs. We need this common flow to be simple on the route side and in the component side. There should be just a Popupmenu which does the right thing for desktop and mobile